cpp/src/arrow/CMakeLists.txt

```diff
@@ -189,6 +189,7 @@ set(ARROW_SRCS
     util/utf8.cc
     util/value_parsing.cc
     vendored/base64.cpp
+    vendored/datetime/ios.mm
```

Comment:
Shouldn't this file only be compiled on iOS targets? At least it seems it doesn't compile properly on our CI builds.

Reply:
I wasn't sure about usage of these files in general -- `ios.h` is included from `tz.cpp` under `#ifdef __APPLE__`, which implies it's being used on macOS as well?

I could make this file iOS-specific, but perhaps some additional updates should be made in order to consolidate `datetime` module and its usage. If that is the case, I'd need some pointers as I'm not familiar with this portion of the codebase and its usage

Reply:
Hmm, then at least make this Apple-specific. `ios.mm` needs an Objective-C++ compiler, which won't be found on all platforms.